syntax = "proto3";

option java_multiple_files = true;
option java_package = "network.beneath.proto";
option java_outer_classname = "BeneathProto";

package proto;

import "engine.proto";

service Gateway {
  rpc ReadRecords(ReadRecordsRequest) returns (ReadRecordsResponse) {}
  rpc WriteRecords(WriteRecordsRequest) returns (WriteRecordsResponse) {}
  rpc GetStreamDetails(StreamDetailsRequest) returns (StreamDetailsResponse) {}
}

message ReadRecordsRequest {
  bytes instance_id = 1;
  string where = 2;
  int32 limit = 3;
}

message ReadRecordsResponse {
  repeated Record records = 1;
}

message WriteRecordsResponse {
}

message StreamDetailsRequest {
  string project_name = 1;
  string stream_name  = 2;
}

message StreamDetailsResponse {
  bytes current_instance_id = 1;
  bytes project_id = 2;
  string project_name = 3;
  string stream_name = 4;
  repeated string key_fields = 5;
  string avro_schema = 6;
  bool public = 7;
  bool external = 8;
  bool batch = 9;
  bool manual = 10;
}
